main.lua

```diff
@@ -1234,22 +1251,103 @@ end
 
 -- takes Location[] https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#location
 -- and renders them to user
-function showLocations(newBufferTitle, lspLocations, labels)
+function showSymbolLocations(newBufferTitle, lspLocations, labels)
+    local symbols = {}
+    local maxLabelLen = 0
+    for i, lspLoc in ipairs(lspLocations) do
+        local fpath = absPathFromFileUri(lspLoc.uri)
+        local lineNumber = lspLoc.range.start.line + 1
+        local columnNumber = lspLoc.range.start.character + 1
+        local labelLen = #labels[i]
+        symbols[i] = {
+            label = labels[i],
+            location = string.format("%s:%d:%d\n", fpath, lineNumber, columnNumber)
+        }
+        if maxLabelLen < labelLen then maxLabelLen = labelLen end
+    end
+
     local bufContents = ""
+    local format = "%-" .. maxLabelLen .. "s # %s"
+    for _, sym in ipairs(symbols) do
+        bufContents = bufContents .. string.format(format, sym.label, sym.location)
+    end
+
+    local newBuffer = buffer.NewBuffer(bufContents, newBufferTitle)
+    newBuffer.Type.Scratch = true
+    newBuffer.Type.Readonly = true
+    micro.CurPane():HSplitBuf(newBuffer)
+end
+
+-- takes Location[] https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#location
+-- and renders them to user
+function showReferenceLocations(newBufferTitle, lspLocations)
+    local references = {}
     for i, lspLoc in ipairs(lspLocations) do
         local fpath = absPathFromFileUri(lspLoc.uri)
         local lineNumber = lspLoc.range.start.line + 1
         local columnNumber = lspLoc.range.start.character + 1
-        local line = string.format("%s:%d:%d\n", fpath, lineNumber, columnNumber)
-        if labels ~= nil then
-            line = labels[i] .. "\t" .. line
+        references[i] = {
+            path = fpath,
+            line = lineNumber,
+            column = columnNumber,
+        }
+    end
+
+    table.sort(references, function(a, b)
+        if a.path < b.path then return true
+        elseif a.path > b.path then return false
+        else
+            if a.line < b.line then return true
+            elseif a.line > b.line then return false
+            else
+                if a.column < b.column then return true
+                elseif a.column > b.column then return false
+                else return true end
+            end
```

Comment:
This can be simplified:
```suggestion
    table.sort(references, function(a, b)
        if a.path ~= b.path then return a.path < b.path end
        if a.line ~= b.line then return a.line < b.line end
        return a.column < b.column
```